benchexec/tablegenerator/react-table/src/components/QuantilePlot.js

```diff
@@ -232,67 +307,75 @@ export default class QuantilePlot extends React.Component {
     }
   };
 
-  renderData = (column, toolIdx, field) => {
+  renderData = (colTitle, toolIdx, field) => {
     const isOrdinal = this.handleType() === "ordinal";
-    let arrayY = [];
     const colIdx = this.props.tools[toolIdx].columns.findIndex(
-      (value) => value.display_title === column,
+      (value) => value.display_title === colTitle,
     );
+    let arrayY = [];
+    let scoreOfIncorrectResults = 0;
 
     if (
       !this.state.isValue ||
-      (colIdx >= 0 && !this.props.hiddenCols[toolIdx].includes(colIdx))
+      (colIdx >= 0 && this.isColVisible(toolIdx, colIdx))
     ) {
       arrayY = this.props.table.map((runSet) => {
         // Get y value if it should be shown and normalize it.
         // For correct x values, arrayY needs to have same length as table.
         const runResult = runSet.results[toolIdx];
         let value = null;
         if (
-          this.state.results !== resultsOptions.correct ||
-          runResult.category === "correct"
+          runResult.category === "correct" ||
+          (!this.state.isResultSelectionDisabled &&
+            this.state.results !== this.resultsOptions.correct)
         ) {
-          value = runResult.values[colIdx].raw;
-          if (value === undefined) {
-            value = null;
-          }
+          value = runResult.values[colIdx].raw || null;
           if (!isOrdinal && value !== null) {
-            value = +value;
-            if (!isFinite(value)) {
-              value = null;
-            }
+            value = isFinite(+value) ? +value : null;
           }
+        } else if (
+          this.state.plot === this.plotOptions.scoreBased &&
+          runResult.score &&
+          runResult.category !== "correct"
+        ) {
+          scoreOfIncorrectResults += runResult.score;
         }
-        return [value, this.props.getRowName(runSet)];
+        return {
+          value,
+          rowName: this.props.getRowName(runSet),
+          score: runResult.score,
+        };
       });
 
-      if (this.state.plot === plotOptions.quantile) {
-        arrayY = arrayY.filter((element) => element[0] !== null);
-        arrayY = this.sortArray(arrayY, column);
+      if (this.state.plot !== this.plotOptions.direct) {
+        arrayY = arrayY.filter((dataObj) => dataObj.value !== null);
+        arrayY = this.sortArray(arrayY, colTitle);
       }
     }
 
     this.hasInvalidLog = false;
     const newArray = [];
-
-    arrayY.forEach((el, i) => {
-      const value = el[0];
+    let xPositionCounter = scoreOfIncorrectResults;
```

Comment:
I guess it would be easier to understand if the code is refactored and `xPositionCounter` is always used for quantile plots? I.e., initialize with 0 or `scoreOfIncorrectResults` and then increment by 1 or score? Then we can get rid of `i` and do not need the nested `+=` as side effect.